sdk/go/internal/querybuilder/marshal.go

```diff
@@ -45,7 +59,13 @@ func marshalValue(ctx context.Context, v reflect.Value) (string, error) {
 	case reflect.Int:
 		return fmt.Sprintf("%d", v.Int()), nil
 	case reflect.String:
-		return fmt.Sprintf("%q", v.String()), nil
+		name := t.Name()
+		// distinguish enum const values and customScalars from string type
+		_, found := customScalar[t.Name()]
+		if name != "string" && !found {
```

Comment:
This took a moment to grok, but it makes sense. We represent enums as strings in Go, whose string value is the GraphQL constant literal syntax like `FOO`, so we want to render that instead of `"FOO"`.

This is as opposed to regular non-enum scalars which can be represented by a string literal. I guess GraphQL will complain if you try to put a string literal in place of an enum?

Reply:
Exactly 👍

Reply:
Exactly! That's what lead us to that not so obvious hack.